### ISSUE TITLE
ecdh_tmp_auto member of struct cert_st is referenced in tls1_get_curvelist without covered by OPENSSL_NO_ECDH ifdef

### ISSUE THREAD

**Issue by vladak** [openssl/openssl#3302]
Trying to compile `OpenSSL_1_0_2-stable` branch after running `Configure` with `no-ecdh` fails with:

```
t1_lib.c: In function 'tls1_get_curvelist':
t1_lib.c:503:42: error: 'struct cert_st' has no member named 'ecdh_tmp_auto'
```

This is because the structure defintion contains this:

```
629  # ifndef OPENSSL_NO_ECDH
630      EC_KEY *ecdh_tmp;
631      /* Callback for generating ephemeral ECDH keys */
632      EC_KEY *(*ecdh_tmp_cb) (SSL *ssl, int is_export, int keysize);
633      /* Select ECDH parameters automatically */
634      int ecdh_tmp_auto;
635  # endif
```

and the code in `tls1_get_curvelist` looks like this:

```
503                  if (!s->server || s->cert->ecdh_tmp_auto) {
504                      *pcurves = eccurves_auto;
505                      pcurveslen = sizeof(eccurves_auto);
506                  } else {
507                      *pcurves = eccurves_all;
508                      pcurveslen = sizeof(eccurves_all);
509                  }
510              }
```

i.e. is not covered by the `OPENSSL_NO_ECDH` ifdef.

**Comment by vladak** [comment]
Possibly it would be good to check all the various configure options with Travis ?

**Comment by richsalz** [comment]
That's too many builds for Travis, we overload it.  We do instead have a script on our server that tests most/all of the no-xxx options.  I wonder if it runs in 1.0.2?

**Comment by mattcaswell** [comment]
We only check master and 1.1.0

**Comment by mspncp** [comment]
See pull request #7606 for a fix.

**Comment by mspncp** [comment]
Fixed by #7606.
